src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisWebhookService.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    private String attachmentToBase64(AttachmentUnit attachmentUnit) {
+        Path path = FilePathService.actualPathForPublicPathOrThrow(URI.create(attachmentUnit.getAttachment().getLink()));
+        try {
+            byte[] fileBytes = Files.readAllBytes(path);
+            return Base64.getEncoder().encodeToString(fileBytes);
+        }
+        catch (IOException e) {
+            throw new IrisInternalPyrisErrorException(e.getMessage());
+        }
+    }
+
+    private PyrisLectureUnitWebhookDTO processAttachmentForUpdate(AttachmentUnit attachmentUnit) {
+        Long lectureUnitId = attachmentUnit.getId();
+        String lectureUnitName = attachmentUnit.getName();
+        Long lectureId = attachmentUnit.getLecture().getId();
+        String lectureTitle = attachmentUnit.getLecture().getTitle();
+        Long courseId = attachmentUnit.getLecture().getCourse().getId();
+        String courseTitle = attachmentUnit.getLecture().getCourse().getTitle();
+        String courseDescription = attachmentUnit.getLecture().getCourse().getDescription() == null ? "" : attachmentUnit.getLecture().getCourse().getDescription();
+        String base64EncodedPdf = attachmentToBase64(attachmentUnit);
+        return new PyrisLectureUnitWebhookDTO(true, artemisBaseUrl, base64EncodedPdf, lectureUnitId, lectureUnitName, lectureId, lectureTitle, courseId, courseTitle,
+                courseDescription);
+    }
+
+    private PyrisLectureUnitWebhookDTO processAttachmentForDeletion(AttachmentUnit attachmentUnit) {
+        Long lectureUnitId = attachmentUnit.getId();
+        Long lectureId = attachmentUnit.getLecture().getId();
+        Long courseId = attachmentUnit.getLecture().getCourse().getId();
+        return new PyrisLectureUnitWebhookDTO(false, artemisBaseUrl, "", lectureUnitId, "", lectureId, "", courseId, "", "");
+    }
+
+    /**
+     * send the updated / created attachment to Pyris for ingestion if autoLecturesUpdate is enabled
+     *
+     * @param courseId               Id of the course where the attachment is added
+     * @param newAttachmentUnits     the new attachment Units to be sent to pyris for ingestion
+     * @param irisSettingsRepository the repository to get the course settings from
+     * @return true if the units were sent to pyris
+     */
+    public boolean autoUpdateAttachmentUnitsInPyris(IrisSettingsRepository irisSettingsRepository, Long courseId, List<AttachmentUnit> newAttachmentUnits) {
+        IrisCourseSettings courseSettings = irisSettingsRepository.findCourseSettings(courseId).isPresent() ? irisSettingsRepository.findCourseSettings(courseId).get() : null;
+        if (courseSettings != null && courseSettings.getIrisLectureIngestionSettings() != null && courseSettings.getIrisLectureIngestionSettings().isEnabled()
+                && Boolean.TRUE.equals(courseSettings.getIrisLectureIngestionSettings().getAutoIngestOnLectureAttachmentUpload())) {
```

Comment:
Since the method now returns `boolean`
```suggestion
                && courseSettings.getIrisLectureIngestionSettings().getAutoIngestOnLectureAttachmentUpload()) {
```